cli/lsp/diagnostics.rs

```diff
@@ -301,68 +301,46 @@ fn ts_json_to_diagnostics(
     .collect()
 }
 
-// Returns `ConfigSnapshot::root_uri` in the correct format.
-fn get_root_specifier(
-  snapshot: &language_server::StateSnapshot,
-) -> Option<ModuleSpecifier> {
-  let root = snapshot.config.root_uri.as_ref()?;
-  let root = root.to_file_path().ok()?;
-
-  // FIXME: `root_uri` from `ConfigSnapshot` are without a trailing slash,
-  // so `Url::join` treats the root as a file, not a directory, and erases the folder name.
-  // To fix that behaviour we just parsing `root_uri` again.
-  ModuleSpecifier::from_directory_path(root).ok()
-}
-
 // Filters documents according to the `include` and the `exclude` lists (from `StateSnapshot::maybe_lint_config`).
 // If a document is in the `exclude` list - then it be removed.
 // If the `include` list is not empty, and a document is not in - then it be removed too.
-fn filter_documents(
+fn filter_lint_documents(
   snapshot: &language_server::StateSnapshot,
   documents: &mut Vec<Document>,
 ) {
-  let root_uri = match get_root_specifier(snapshot) {
-    Some(uri) => uri,
-    None => return,
-  };
-
-  let linter_config = match &snapshot.maybe_lint_config {
+  let lint_config = match &snapshot.maybe_lint_config {
     Some(config) => config,
     None => return,
   };
 
-  let join_specifiers = |specifiers: &Vec<String>| -> Vec<ModuleSpecifier> {
-    specifiers
-      .iter()
-      .filter_map(|i| root_uri.join(i).ok())
-      .collect()
-  };
-
-  let ignore_specifiers = join_specifiers(&linter_config.files.exclude);
-  let include_specifiers = join_specifiers(&linter_config.files.include);
-
   documents.retain(|doc| {
-    let path = doc.specifier().path();
+    let path = if let Ok(file_path) = doc.specifier().to_file_path() {
```

Comment:
The tests are hanging on Windows because of this if expression returning false. That's happening because the tests are using file specifiers that the `url` crate doesn't resolve to a `Path` though I wish it did (ex. `file:///test.ts`... I opened https://github.com/servo/rust-url/issues/730). I see a couple options:

1. We add a new `to_file_path(specifier: &ModuleSpecifier)` function that does more of an attempt to convert `file:///something` to `PathBuf` on Windows.
2. We change all the tests to use platform specific paths (seems error prone... would rather the code would just handle it on Windows too).

Reply:
I thought we had something like `to_file_path()` that failed friendly on Windows, I will try to dig it out.

Reply:
I did some more searching and didn't find anything so I added a `specifier_to_file_path` function just now.